server/src/infra/graphql/mutation.rs

```diff
@@ -163,6 +163,48 @@ fn unpack_attributes(
     })
 }
 
+fn consolidate_attributes(
+    attributes: Vec<AttributeValue>,
+    first_name: Option<String>,
+    last_name: Option<String>,
+    avatar: Option<String>,
+) -> Vec<AttributeValue> {
+    // Prepare map with the attributes from fields, which will have lower
+    // priority than a value from a corresponding attribute already given.
+    let mut fallback_attrs: BTreeMap<String, AttributeValue> = BTreeMap::new();
+    if let Some(first_name) = first_name {
+        let attr_value = AttributeValue {
+            name: "first_name".into(),
+            value: vec![first_name],
+        };
+        fallback_attrs.insert(attr_value.name.clone(), attr_value);
+    }
+    if let Some(last_name) = last_name {
+        let attr_value = AttributeValue {
+            name: "last_name".into(),
+            value: vec![last_name],
+        };
+        fallback_attrs.insert(attr_value.name.clone(), attr_value);
+    }
+    if let Some(avatar) = avatar {
+        let attr_value = AttributeValue {
+            name: "avatar".into(),
+            value: vec![avatar],
+        };
+        fallback_attrs.insert(attr_value.name.clone(), attr_value);
+    }
+    // Prepare map of the client provided attributes
+    let mut provided_attributes: BTreeMap<String, AttributeValue> = attributes
```

Comment:
Maybe simpler: start with this map, and   insert the value of it's not already there in the loop. The logic will be clearer as well

Reply:
Done.